### PR TITLE
#1598 #1594 #1591 #1508

### PR DESCRIPTION
**Pull request by emimaesmith**
##### Checklist

- [x] `manage.py test` passes
- [x] tests are included
- [x] documentation is changed or added

### Affected app(s)

batch, diagnostic, error, ingest, job, metrics, node, queue, recipe, scheduler, storage

### Description of change

#1598 Removed the requirement for the recipe revision number from the strike/scan configuration. Revision number is optional and the latest will be used if not specified.
#1594 Fixed bug where updating a recipe type (causing an updated revision number) broke the strike/scan configuration due to a recipe not found error
#1591 Removed leftover old trigger references
#1508 Added v7 API calls to the views

### CODE REVIEW

**Comment by emimaesmith** [comment]
Ready for review. I can include specific tests for v7 API calls, but since functionality is exactly the same as v6, I didn't think it was completely necessary to duplicate test cases.

File: scale/diagnostic/views.py [hunk 1]
```diff
@@ -23,7 +23,16 @@ class QueueScaleBakeView(GenericAPIView):
...
+    
+    def get_serializer_class(self):
+        """Returns the appropriate serializer based off the requests version of the REST API. """
+
+        if self.request.version == 'v6':
+            return QueueStatusSerializerV6
+        elif self.request.version == 'v7':
+            return QueueStatusSerializerV6
+        else:
+            return QueueStatusSerializer
```
Comment (JohnPTobe) [inline scale/diagnostic/views.py]: We should get rid of this else and remove the queuestatusserializer class

File: scale/diagnostic/views.py [hunk 2]
```diff
@@ -76,7 +87,16 @@ class QueueScaleCasinoView(GenericAPIView):
...
+
+    def get_serializer_class(self):
+        """Returns the appropriate serializer based off the requests version of the REST API. """
+
+        if self.request.version == 'v6':
+            return QueueStatusSerializerV6
+        elif self.request.version == 'v7':
+            return QueueStatusSerializerV6
+        else:
+            return QueueStatusSerializer
```
Comment (JohnPTobe) [inline scale/diagnostic/views.py]: same as above

File: scale/diagnostic/views.py [hunk 3]
```diff
@@ -129,7 +151,16 @@ class QueueScaleHelloView(GenericAPIView):
...
```
Comment (JohnPTobe) [inline scale/diagnostic/views.py]: same

File: scale/queue/views.py [hunk 4]
```diff
@@ -50,6 +50,10 @@ def get_serializer_class(self):
 
         if self.request.version == 'v6':
             return QueueStatusSerializerV6
+        elif self.request.version == 'v7':
+            return QueueStatusSerializerV6
+        else:
+            return QueueStatusSerializer
```
Comment (JohnPTobe) [inline scale/queue/views.py]: remove queuestatusserializer class

File: scale/diagnostic/views.py (hunk 1 above)
Comment (emimaesmith) [reply scale/diagnostic/views.py]: Removed

File: scale/diagnostic/views.py (hunk 3 above)
Comment (emimaesmith) [reply scale/diagnostic/views.py]: Removed

File: scale/queue/views.py (hunk 4 above)
Comment (emimaesmith) [reply scale/queue/views.py]: Removed

File: scale/diagnostic/views.py (hunk 2 above)
Comment (emimaesmith) [reply scale/diagnostic/views.py]: Removed

**Pull request merged**